js/battledata.js

```diff
@@ -1024,6 +1025,18 @@ var Tools = {
 				}
 				template.formeid = formeid;
 			}
+			if (template.otherForms) {
+				var form = '', formid = '';
+				if (template.otherForms.indexOf(speciesid) >= 0) {
+					form = speciesid.slice(template.baseSpecies.length);
+					formid = '-' + form;
+					form = form[0].toUpperCase() + form.slice(1);
+				}
+				template.form = form;
```

Comment:
A shallow clone is a clone that's only one level deep.

e.g. `obj2 = $.merge({}, obj1)` is the usual way to create a shallow clone.

and then if you do `obj2.num = 2`, it won't change `obj1.num`

but if you do `obj2.point.x = 1`, it will change `obj1.point.x`.